src/app/campaign/store/reducers/flight.reducer.ts

```diff
@@ -137,11 +139,25 @@ const _reducer = createReducer(
       { id: flight.id, changes: { localFlight: { ...localFlight, allocationStatus, allocationStatusMessage } } },
       state
     );
+  }),
+  on(campaignActions.CampaignFlightZoneAddCreatives, (state, action) => {
+    const { flightId, zoneId, creativeIds } = action;
+    const localFlight = state.entities[flightId] && state.entities[flightId].localFlight;
+    const creativeFlightZones = creativeIds.map(creativeId => ({ creativeId }));
+    const zones =
+      localFlight &&
+      localFlight.zones.map(zone =>
+        zone.id === zoneId
+          ? {
+              ...zone,
+              creativeFlightZones: zone.creativeFlightZones ? [...zone.creativeFlightZones, ...creativeFlightZones] : creativeFlightZones
```

Comment:
I went through an iteration with this where the creatives were including in the flight zones and I was just storing them here as they came in so I could use the creative filename for the card. I had to do some "creative" effects handling to get those creatives loaded with the flights, but it's much better. I also considering moving flight zones to a separate reducer or at least changing the shape of them here so they are keyed by id instead of an array, but I decided to stop where I had things and get this done.